frontend/src/components/nodes/NodesMap.tsx

```diff
@@ -36,83 +33,38 @@ interface State {
+    nodesType: "",
     newNodes: [],
     removedNodes: [],
     nodeClusters: [],
   };
 
   componentDidMount() {
-    const { nodeInfo } = this.context;
-    this.fetchGeo(nodeInfo.validators, nodeInfo.onlineNodes);
+    this.fetchGeo();
   }
 
   // need to change the method to nodes.js
-  fetchGeo = async (validators: any, onlineNodes: any) => {
-    const nodes =
-      this.state.nodesType === "validators" ? validators : onlineNodes;
-    const url =
-      "http://ip-api.com/batch?fields=status,message,country,city,latitude,longitude,timezone,query";
-    if (nodes) {
-      const IPsArray = nodes.map((node: any) => node.ipAddress);
-      let ipCount = IPsArray.length;
-      let geoData: IGeo[] = [];
-      while (ipCount > 0) {
-        let min = Math.max(ipCount - 50, 0);
-        let ipArrPart = JSON.stringify(IPsArray.slice(min, ipCount));
-        let response = await fetch(url, {
-          method: "POST",
-          mode: "cors",
-          cache: "no-cache",
-          credentials: "same-origin",
-          headers: {
-            "Content-Type": "application/json",
-          },
-          redirect: "follow",
-          referrerPolicy: "no-referrer",
-          body: ipArrPart,
-        });
-        let part = await response.json();
-        geoData = geoData.concat(part);
-        ipCount -= 50;
-      }
-
-      const bubbles: IBubble[] = nodes.map((element: any, index: number) => {
-        const bubble: IBubble = {
-          geo: {
-            latitude: "",
-            longitude: "",
-            city: "",
-          },
-          radius: 0,
-          nodeInfo: {
-            ipAddress: "",
-            accountId: "",
-            nodeId: "",
-            lastSeen: 0,
-            lastHeight: 0,
-            agentName: "",
-            agentVersion: "",
-            agentBuild: "",
-            status: "",
-          },
-        };
-
-        bubble.geo.latitude = geoData[index].latitude;
-        bubble.geo.longitude = geoData[index].longitude;
-        bubble.geo.city = geoData[index].city;
-        bubble.radius = 4;
-        if (this.state.nodesType === "validators") {
-          bubble.nodeInfo = element.nodeInfo;
-          bubble.nodeInfo.accountId = element.account_id;
-        }
-        bubble.nodeInfo = element;
+  fetchGeo = async () => {
+    const { nodeInfo } = this.context;
+    let validators = nodeInfo.validators.filter(
+      (node: N.Validating) => node.new !== true
+    );
 
-        return bubble;
-      });
+    const nodes =
+      this.state.nodesType === "validators" ? validators : nodeInfo.onlineNodes;
+
+    const bubbles: IBubble[] = nodes.map((node: N.NodeInfo & N.Validating) => {
+      return {
+        radius: 4,
+        fillKey: "validatorBubbleFill",
+        nodeInfo: node,
+        nodeId: node.nodeId || node.nodeInfo?.nodeId || "Unknown",
+        latitude: node.latitude || node.nodeInfo?.latitude,
+        longitude: node.longitude || node.nodeInfo?.longitude,
```

Comment:
Can we improve the data layout so we don't need to do all these hops and dances?

Reply:
if we change here to the same format, we will change backend with this to match the same format.

Reply:
see here: https://github.com/near/near-explorer/blob/7356915e75d9c937c8fe49335ef1f26456bcc6d7/backend/src/db-utils.js#L63 
the online node and validating node have different attrs, so it is different format.